skeleton/src/main/java/com/dropbox/kaiken/skeleton/skeleton/core/KaikenUserServicesProvider.kt

```diff
@@ -0,0 +1,67 @@
+package com.dropbox.kaiken.skeleton.skeleton.core
+
+import com.dropbox.kaiken.scoping.AppServices
+import com.dropbox.kaiken.scoping.TeardownHelper
+import com.dropbox.kaiken.scoping.UserServices
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * Skeleton implementation of [UserServicesProvider] that registers itself with a [UserManager].
+ *
+ */
+class KaikenUserServicesProvider
+constructor(
+        private val applicationServices: AppServices,
+        private val userServicesFactory: (AppServices, SkeletonUser) -> UserServices,
+) : SkeletonUserServicesProvider {
+    private val userServicesMap = mutableMapOf<String, KaikenUserServices>()
+
+    init {
+        //TODO Mike figure out if we want another scope to launch from
+        GlobalScope.launch {
```

Comment:
I'm guessing that the reason you're doing `GlobalScope.launch` is because you don't want to block. Which makes. But at some point you're going to have to block (i.e. you can't dismiss the splash until the UserManager and the UserServices for the active user have been initialized). At some point there's has to be a blocking operation. We should make explicit, where that blocking operation happens.

Reply:
Yup! Another area @chris-mitchell is working on, how to block on the first read. There are times we don't have to go into a logged in state and can delay the blocking for a bit